pages/index.js

```diff
@@ -2,64 +2,45 @@ import styled from "styled-components";
 import { useEffect, useState } from "react";
 import Image from "next/image";
 
-
-
-
-
-export default function Home() {
-
-  const [data, setData] = useState();
-
-
-  const url = 'https://imdb8.p.rapidapi.com/auto-complete?q=marvel';
+function MoviesList() {
+  const [movies, setMovies] = useState([]);
 
   useEffect(() => {
-    const options = {
-      method: 'GET',
-      headers: {
-        'X-RapidAPI-Key': '467e3c9b58mshc5c2a10984fd07bp15ceebjsn338c8a8d0154',
-        'X-RapidAPI-Host': 'imdb8.p.rapidapi.com'
-      }
-    };
-
-fetch(url, options)
-.then(response => response.json())
-.then(response => {
-  console.log(response);
-  setData(response);
-})
-.catch(err => {
-  console.error(err);
-})
-
-  })
+    async function fetchMovies() {
+      const apiKey = process.env.API_KEY;
+      const response = await fetch(
+        `https://api.themoviedb.org/3/movie/popular?api_key=9b1dc3e910cce9037e3a21dbbd912753&language=en-US&page=1`
+      );
```

Comment:
You're reading the key from the environment object, but you're not using that information in your fetch. Instead, you're passing the secret key itself. Better use the `apiKey` here.